PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/ConnectionStringCheck.java

```diff
@@ -0,0 +1,44 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+
+import com.intellij.codeInspection.*;
+import com.intellij.psi.*;
+import com.microsoft.applicationinsights.TelemetryClient;
+import org.jetbrains.annotations.NotNull;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class extends the LocalInspectionTool and checks for the usage of connection strings in the code.
+ * Connection strings that start with "DefaultEndpointsProtocol=http;AccountName=" are flagged as problems.
+ * This is because the use of such connection strings is not recommended.
+ */
+public class ConnectionStringCheck extends LocalInspectionTool {
+
+    @NotNull
+    @Override
+    public PsiElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
+
+        TelemetryClient telemetry = TelemetryClientProvider.getTelemetryClient();
+
+        return new JavaElementVisitor() {
+
+            @Override
+            public void visitLiteralExpression(PsiLiteralExpression expression) {
+                super.visitLiteralExpression(expression);
+
+                String value = expression.getValue() instanceof String ? (String) expression.getValue() : null;
+
+                if (value != null && value.startsWith("DefaultEndpointsProtocol=http;AccountName=")) {
```

Comment:
Will this logic end up becoming repetitive as you expand it to cover more services than just storage? You might want to consider how you foresee this coming together and decide if it's worth planning your implementation to be expandable now, so you don't have to refactor a bunch of logic later.

Reply:
Expanding on Jair's comment, an approach could be having a collection of known ways a connection string can start and iterate through it, for example.

Reply:
refactored it to check for use of connectionString API call, so no need to track presence of a connection string anymore:) this works for all services